packages/flutter/lib/src/widgets/selection.dart

```diff
@@ -0,0 +1,586 @@
+    }
+  }
+
+  void _endDragSelection() {
+    _currentDragPosition = null;
+    _scheduledSelectionUpdate = false;
+    _selectionUpdater.dispatchSelectionEvent(const DragSelectionEndEvent());
+  }
+
+  void _cancelSelection() {
+    _endDragSelection();
+    _selectionUpdater.clear();
+  }
+
+  void _selectAll(Intent intent) {
+    _cancelSelection();
+    _selectionUpdater.selectAll();
+  }
+
+  Future<void> _copy(Intent intent) async {
+    final Object? data = _selectionUpdater.copy();
+    if (data == null) {
+      return;
+    }
+    Clipboard.setData(ClipboardData(text: data as String));
+  }
+
+  void _handleSelectablesChanged(List<Selectable> selectables) {
+    _selectionUpdater.updateSelectables(selectables);
+  }
+
+  @override
+  void dispose() {
+    _selectionUpdater.clear();
+    _focusNode.dispose();
+    super.dispose();
+  }
+
+  Map<ShortcutActivator, Intent> _getPlatformShortcuts() {
+    switch (defaultTargetPlatform) {
+      case TargetPlatform.iOS:
+      case TargetPlatform.android:
+      case TargetPlatform.fuchsia:
+      case TargetPlatform.linux:
+      case TargetPlatform.windows:
+        return _kWindowsShortcuts;
+
+      case TargetPlatform.macOS:
+        return _kMacShortcuts;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Shortcuts(
+      debugLabel: 'Selection Area shortcuts',
+      shortcuts: _getPlatformShortcuts(),
+      child: Actions(
+        actions: _actions,
+        child: Focus(
+          focusNode: _focusNode,
+          child: RawGestureDetector(
+            gestures: _gestureRecognizers,
+            behavior: HitTestBehavior.translucent,
+            excludeFromSemantics: true,
+            child: SelectionContainer(
+              enabled: true,
+              onSelectablesChange: _handleSelectablesChanged,
+              child: widget.child,
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _SelectAllIntent extends Intent {
+  const _SelectAllIntent();
+}
+
+class _CopyIntent extends Intent {
+  const _CopyIntent();
+}
+
+/// An abstract base class for updating multiple selectable children.
+///
+/// This class optimize the selection update by keeping track the current
+/// selectable that contains the selection edge.
+abstract class MultiSelectableSelectionUpdaterBase {
+
+  /// Gets the list selectables this updater is responsible for updating.
+  ///
+  /// The subclasses is responsible for updating the selection when the
+  /// selectables in the list has changed.
+  List<Selectable> get selectables;
+
+  /// The current selectable that contains the selection edge.
+  @protected
+  int currentSelectionIndex = -1;
+
+  /// Removes the selection of all the selectables this updater managed.
+  void clear() {
+    for (final Selectable selectable in selectables) {
+      selectable.clearSelection();
+    }
+    currentSelectionIndex = -1;
+  }
+
+  /// Copies the selected contents of all the selectables.
+  Object? copy() {
```

Comment:
Should this be typed `String?`? Seems to be the only possible return type?

Reply:
It may not be the case if we support multimedia, but i guess we can change that later?